src/query/expression/src/evaluator.rs

```diff
@@ -195,6 +208,100 @@ impl<'a> Evaluator<'a> {
         result
     }
 
+    fn run_udf_server_call(
+        &self,
+        func_name: &str,
+        server_addr: &str,
+        return_type: &DataType,
+        args: &[Expr],
+        validity: Option<Bitmap>,
+    ) -> Result<Value<AnyType>> {
+        let inputs = args
+            .iter()
+            .map(|expr| self.partial_run(expr, validity.clone()))
+            .collect::<Result<Vec<_>>>()?;
+        assert!(
+            inputs
+                .iter()
+                .filter_map(|val| match val {
+                    Value::Column(col) => Some(col.len()),
+                    Value::Scalar(_) => None,
+                })
+                .all_equal()
+        );
+
+        // construct input record_batch
+        let num_rows = self.input_columns.num_rows();
+        let fields = args
+            .iter()
+            .enumerate()
+            .map(|(idx, arg)| DataField::new(&format!("arg{}", idx + 1), arg.data_type().clone()))
+            .collect_vec();
+        let data_schema = DataSchema::new(fields);
+
+        let block_entries = inputs
+            .into_iter()
+            .zip(args.iter())
+            .map(|(col, arg)| {
+                let arg_type = arg.data_type().clone();
+                let block = if contains_variant(&arg_type) {
+                    BlockEntry::new(arg_type, transform_variant(&col, true)?)
+                } else {
+                    BlockEntry::new(arg_type, col)
+                };
+                Ok(block)
+            })
+            .collect::<Result<Vec<_>>>()?;
+
+        let input_batch = DataBlock::new(block_entries, num_rows)
+            .to_record_batch_keep_schema(&data_schema)
+            .map_err(|err| ErrorCode::from_string(format!("{err}")))?;
+
+        let func_name = func_name.to_string();
+        let server_addr = server_addr.to_string();
+        let result_batch = GlobalQueryRuntime::instance()
```

Comment:
We should make `run_udf_server_call` async, or it's possible to make threads hang and can't be executed by batch.  Suggestions from @zhang2014

Reply:
Let's do it in another PR